msvc/scripts/genproj.cs

```diff
@@ -71,18 +72,23 @@ public void Add (MsbuildGenerator.VsCsproj vsproj)
 		try {
 			libraries.Add (vsproj);
 		} catch (Exception ex) {
-			Console.WriteLine (ex);
+			Console.Error.WriteLine ($"Error while adding library: {ex.Message}");
 		}
 	}
 
-	private void WriteProjectReference (StreamWriter sln, string prefixGuid, string library, string relativePath, string projectGuid, params string[] dependencyGuids)
+	private void WriteProjectReference (StreamWriter sln, string prefixGuid, string library, string relativePath, string projectGuid, string[] dependencyGuids)
 	{
+		// HACK
+		library = library.Replace("-net_4_x", "");
 		sln.WriteLine (project_start, prefixGuid, library, relativePath, projectGuid);
 
-		foreach (var guid in dependencyGuids) {
-    		sln.WriteLine ("    ProjectSection(ProjectDependencies) = postProject");
-    		sln.WriteLine ("        {0} = {0}", guid);
-    		sln.WriteLine ("    EndProjectSection");
+		if (
+			(dependencyGuids != null) && (dependencyGuids.Length > 0)
+		) {
+			sln.WriteLine ("\tProjectSection(ProjectDependencies) = postProject");
+			foreach (var guid in dependencyGuids)
+	    		sln.WriteLine ("\t\t{0} = {0}", guid);
```

Comment:
This looks suspicious, why is the same thing assign from and to?

Reply:
I don't know, .sln is a really bad mysterious format. I discovered using spaces instead of tabs causes silent parsing errors :(